src/services/EcoBridge/EcoBridgeProvider.tsx

```diff
@@ -0,0 +1,40 @@
+import React, { useContext, useEffect, useState } from 'react'
+import { EcoBridge } from './EcoBridge'
+import store from '../../state'
+import { useActiveWeb3React } from '../../hooks'
+import { ecoBridgeConfig } from './EcoBridge.config'
+
+export const EcoBridgeContext = React.createContext<EcoBridge | null>(null)
+
+export const EcoBridgeProvider = ({ children }: { children?: React.ReactNode }) => {
+  const { library, account, chainId } = useActiveWeb3React()
+  const [ecoBridge, setEcoBridge] = useState<EcoBridge | null>(null)
+
+  useEffect(() => {
+    const initEcoBridge = async () => {
+      if (!ecoBridge) {
+        const ecoBridgeInstance = new EcoBridge(store, ecoBridgeConfig)
+        setEcoBridge(ecoBridgeInstance)
+      }
+
+      if (ecoBridge && account && library && chainId) {
+        if (!ecoBridge.ready) {
+          await ecoBridge.init({ account, activeChainId: chainId, activeProvider: library })
+        } else {
+          await ecoBridge.updateSigner({ account, activeChainId: chainId, activeProvider: library })
+        }
+      }
+    }
+    initEcoBridge()
+  }, [account, chainId, library, ecoBridge])
+
+  return <EcoBridgeContext.Provider value={ecoBridge}>{children}</EcoBridgeContext.Provider>
+}
+
+export const useEcoBridge = () => {
+  const ecoBridge = useContext(EcoBridgeContext)
+  if (!ecoBridge) {
+    throw new Error('No EcoBridge - this shouldnt happen')
+  }
```

Comment:
Correct, but this error won't occur, it's here only to make sure that EcoBride is not optional and we don't need to check it every time before it's used. We could do `return ecoBridge as EcoBridge` and it would work as well but I feel that throwing error gives better dev experience. 

We could also instantiate EcoBridge outside React and then problem would be solved (EcoBridge would have initial context instead of null) but after discussions with Marek we abandoned this idea.

Reply:
Error message improved in #908. This problem is also addressed in https://kentcdodds.com/blog/how-to-use-react-context-effectively